### ISSUE TITLE
Floating point exception can be triggered in AvgPool3D when run with CPU and OneDNN is enbaled

### ISSUE THREAD

**Issue by shijy16** [tensorflow/tensorflow#57500]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9 and 2.11.0-dev20220828

### Custom Code

No

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.10

### CUDA/cuDNN version

CUDA 11.5

### Current Behaviour?

```shell
When run with CPU and OneDNN is enabled, an FPE can be triggered in AvgPool3D, which may result in DoS.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
with tf.device("CPU"):
    input = tf.random.uniform([30, 19, 4, 19, 17], dtype=tf.float32)
    ksize =[1, 13, 3, 20, 1]
    strides = [1, 14, 4, 1, 1]
    padding = "VALID"
    data_format = "NDHWC"
    res = tf.raw_ops.AvgPool3D(
        input=input,
        ksize=ksize,
        strides=strides,
        padding=padding,
        data_format=data_format,
    )
```

### Relevant log output

```shell
Floating point exception (core dumped)
```
</details>

**Comment by tiruk007** [comment]
@shijy16 
I was able to execute the given code without any issue/error on tensorflow-gpu , [cpu v2.9](https://colab.sandbox.google.com/gist/tiruk007/d55fe86d12324feaccfe5a4269f74f76/57500cpu.ipynb#scrollTo=bDoQSBnGR6QG) . kindly find the gist of it [here](https://colab.sandbox.google.com/gist/tiruk007/51eef2c4e0c2bb8d97668bf3a30e7422/57500gpu.ipynb#scrollTo=rnKqzpIYSeZ0).

Thank you!

**Comment by shijy16** [comment]
@tiruk007 
Thanks for response, it seems to be an issue which can only be reproduced when execute on CPU with **OneDNN** enabled. But colab doesnot support **OneDNN**.

**Comment by preethivenkatesh** [comment]
hi @shijy16 The engineering team is able to reproduce your issues and is working on fixing it. By the way, what do you mean by collab doesn't support oneDNN? could you please elaborate?

**Comment by shijy16** [comment]
I mean Google Colab seems to use CPUs which don't support OneDNN. Because when I run code with CPU, I didn't see the following log:
````
oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`
````
And I can't reproduce the issue in Google Colab.

> hi @shijy16 The engineering team is able to reproduce your issues and is working on fixing it. By the way, what do you mean by collab doesn't support oneDNN? could you please elaborate?

**Comment by agramesh1** [comment]
> I mean Google Colab seems to use CPUs which don't support OneDNN. Because when I run code with CPU, I didn't see the following log:
> 
> ```
> oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`
> ```
> 
> And I can't reproduce the issue in Google Colab.

@shijy16  thanks for reporting the issue.  Yes, Google Colab uses older Intel CPUs for which the oneDNN optimizations are not on by default.  To use oneDNN opts with Colab you can set the env variable in the notebook. (It has to be done before the first  import of TensorFlow for it to take effect.
```
import os
os.environ["TF_ENABLE_ONEDNN_OPTS"]="1"
from tensorflow import keras
```

**Comment by sachinprasadhs** [comment]
I was able to reproduce the issue in 2.10.0 version by enabling OneDNN in colab, please find the gist [here](https://colab.sandbox.google.com/gist/sachinprasadhs/25bd523cdda4800460edf313c7b27f1a/untitled73.ipynb).

**Comment by shijy16** [comment]
@agramesh1 Thanks you! I tried `export TF_ENABLE_ONEDNN_OPTS=1` but found that colab doesn't support `export`. Your suggestion helps a lot.

**Comment by cantonios** [comment]
Closed by #57998
